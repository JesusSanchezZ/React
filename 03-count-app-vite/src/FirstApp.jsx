// import { Fragment } from 'react';

// export const FirstApp = () => {
//     return (
//         <Fragment>
//             <h1>First App</h1>
//             <p>Hola a todos</p>
//         </Fragment>
//     );
// }
import PropTypes from 'prop-types';

const newMessage = {
    nombre: 'Jesus',
    mensaje: 'Hola a todos'
};

const saludo = () =>{
    const nombre = newMessage.nombre;
    const mensaje = newMessage.mensaje;

    return(
        <>
            <h1>{nombre} <span>Les dice:</span></h1>
            <p><strong>{mensaje}</strong></p>
        </>
    );
}

const getResult = (a, b) => {
    return a + b;
}

export const FirstApp = ( { title, subTitle, name } ) => {
    //console.log( props );
    
    return(
        <>
            <h1 data-testid="test-title"> { title } </h1>
            {/* <h1>{ JSON.stringify(newMessage) }</h1> */}
            <p>{ subTitle }</p>
            <p>{ subTitle }</p>
            <p>{ name }</p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    //title: 'No hay título',
    subTitle: 'No hay subtítulo',
    name: 'Jesus Sanchez'
}

/**
 * instalar react testing library
 * npm install -D @testing-library/react
 */