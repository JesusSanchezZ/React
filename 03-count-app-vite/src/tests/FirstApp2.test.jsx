import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";

describe('Pruebas en <FirstApp>', ()=>{
    const title    = 'Hola, Soy Jesus';
    const subtitle = 'Soy un subtitulo';

    test('Debe de hacer match con el snapshot',()=>{
        
        render(<FirstApp title={ title }/>);
        expect(screen.getByText(title)).toBeTruthy();
        //screen.debug();
        // expect( container ).toMatchSnapshot();
    });

    test('Debe de mostrar el tÃ­tulo en un h1', ()=>{
        render( <FirstApp title={title}/>);
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subtitulo enviado por props',()=>{
        render(
            <FirstApp
                title={title}
                subTitle= {subtitle}
            />
        );

        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
});