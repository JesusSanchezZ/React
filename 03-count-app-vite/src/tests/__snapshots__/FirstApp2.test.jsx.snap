// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
     
    Hola, Soy Jesus
     
  </div>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Jesus Sanchez
  </p>
</div>
`;
